From b0c98f93a7f427d18206c79bf2a0350dc81172f6 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 6 Apr 2023 10:44:08 -0700
Subject: [PATCH] configure: Replace use of AC_EGREP_CPP

Use AC_COMPILE_IFELSE instead to compute platform support for arp ioctls
AC_EGREP_CPP does not work as used with upcoming autoconf 2.72

Upstream-Status: Submitted [https://github.com/ofalk/libdnet/pull/85]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 m4/acinclude.m4 | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
index d6350a2..3699bf1 100644
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -196,12 +196,11 @@ dnl
 AC_DEFUN([AC_DNET_IOCTL_ARP],
     [AC_MSG_CHECKING(for arp(7) ioctls)
     AC_CACHE_VAL(ac_cv_dnet_ioctl_arp,
-    AC_EGREP_CPP(werd,[#include <sys/types.h>
-                       #define BSD_COMP
-                       #include <sys/ioctl.h>
-                       #ifdef SIOCGARP
-                 werd
-                       #endif],
+    AC_COMPILE_IFELSE(
+        [AC_LANG_PROGRAM([[#include <sys/types.h>
+                           #define BSD_COMP
+                           #include <sys/ioctl.h>]],
+                          [[int foo = SIOCGARP;]])],
     ac_cv_dnet_ioctl_arp=yes,
     ac_cv_dnet_ioctl_arp=no))
     case "$host_os" in
-- 
2.40.0

